### PR TITLE
[Reset Password] Organization Key Pair

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes
- **OrganizationsController**: Pass in public and private keys to the `SignUpAsync` method
- **Public/MemberBaseModel**: Added `ResetPasswordEnrolled` bool to the member response model
- **OrganizationCreateLicenseRequestModel**: Added reference to `OrganizationKeysRequestModel` which will provide the org's public/private key on create
- **OrganizationCreateRequestModel**: Added reference to `OrganizationKeysRequestModel` which will provide the org's public/private key on create
- **OrganizationKeysRequestModel**: Model that will provide the public/private key for an organization. **NOTE** Is there a clean way to simplify the duplicated code for transforming key info to the given objects?
- **OrganizationUpdateRequestModel**: Added reference to `OrganizationKeysRequestModel` which will provide the org's public/private key on update
- **OrganizationUpgradeRequestModel**: Added reference to `OrganizationKeysRequestModel` which will provide the org's public/private key on upgrade
- **OrganizationResponseModel**: Added reference to `UseResetPassword` and `HasPublicAndPrivateKeys`. **NOTE** Of course, open to changing the name of the pub/priv key field. Wasn't sure what direction to go, so I went very literal.
- **PlanResponseModel**: Added reference to `HasResetPassword` that will be translated to `UseResetPassword` on the organization table/models
- **ProfileOrganizationResponseModel**:  Added reference to `UseResetPassword` and `HasPublicAndPrivateKeys`. **NOTE** Of course, open to changing the name of the pub/priv key field. Wasn't sure what direction to go, so I went very literal.
- **OrganizationLicense**: Bumped default version to `7`. Added logic in for `UseResetPassword` validity checks.
- **OrgnizationUpgrade**: Added `PublicKey` and `PrivateKey` fields
- **OrganizationAbility**: Added `UseResetPassword` field
- **OrganizationUserOrganizationDetails**: Added references to `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **StaticStore/Plan**: Added `HasResetPassword` to the plan object
- **Organization**: Added references to `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **IOrganizationService**: Expanded `SignUpAsync` method to include `publicKey` and `privateKey` parameters
- **OrganizationService**: Updated existing methods for saving an organization to include the new `UseResetPassword`, `PublicKey`, and `PrivateKey` fields. Left TODOs for when the policy is in place.
- **Utilities/StaticStore**: Added `HasResetPassword` to the (new) Enterprise plans
- **Organization_Create**: Added references to `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **Organization_ReadAbilities**: Added reference to `UseResetPassword`
- **Organization_Update**: Added references to `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **Tables/Organization**: Added new columns: `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **OrganizationUserOrganizationDetailsView**: Added references to `UseResetPassword`, `PublicKey`, and `PrivateKey`
- **OrgResetPasswordAbilityAndRsaKeys**: Created migrator script to handle all of the necessary db changes

### CODE REVIEW

File: src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using Bit.Core.Models.Table;
+using System.ComponentModel.DataAnnotations;
+using Bit.Core.Models.Business;
+
+namespace Bit.Core.Models.Api
+{
+    public class OrganizationKeysRequestModel
+    {
+        public string PublicKey { get; set; }
+        [Required]
```
Comment (kspearrin) [inline src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs:10]: why only the private key required?

Comment (vincentsalucci) [reply src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs:10]: That's a great question. That's how the `KeysRequestModel` is structured, but I'm not sure if that's by mistake or on purpose.

Comment (vincentsalucci) [reply src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs:10]: @kspearrin Do you see any issue with me adding a `required` decorator to the `PublicKey` as well?

Comment (kspearrin) [reply src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs:10]: No. You can then avoid alot of those validators, i think

Comment (vincentsalucci) [reply src/Core/Models/Api/Request/Organizations/OrganizationKeysRequestModel.cs:11]: Updated!

**Review by kspearrin — APPROVED** [review]

**Pull request merged**